File: Level 1/동영상 재생기/solution.py
def time_to_seconds(time):
    minutes, seconds = time.split(":")
    return int(minutes) * 60 + int(seconds)


def seconds_to_time(seconds):
    return f"{str(seconds // 60).rjust(2, '0')}:{str(seconds % 60).rjust(2, '0')}"


def solution(video_len, pos, op_start, op_end, commands):
    video_len = time_to_seconds(video_len)
    pos = time_to_seconds(pos)
    op_start = time_to_seconds(op_start)
    op_end = time_to_seconds(op_end)

    if op_start <= pos <= op_end:
        pos = op_end

    for command in commands:
        new_pos = pos + 10 if command == "next" else pos - 10

        if new_pos > video_len:
            new_pos = video_len

        if new_pos < 0:
            new_pos = 0

        if op_start <= new_pos <= op_end:
            new_pos = op_end

        pos = new_pos

    return seconds_to_time(pos)

File: Level 1/동영상 재생기/solution.js
function solution(videoLen, pos, opStart, opEnd, commands) {
  videoLen = timeToSeconds(videoLen);
  pos = timeToSeconds(pos);
  opStart = timeToSeconds(opStart);
  opEnd = timeToSeconds(opEnd);

  if (pos >= opStart && pos <= opEnd) {
    pos = opEnd;
  }

  commands.forEach((command) => {
    let newPos = command === "next" ? pos + 10 : pos - 10;

    if (newPos > videoLen) {
      newPos = videoLen;
    }

    if (newPos < 0) {
      newPos = 0;
    }

    if (newPos >= opStart && newPos <= opEnd) {
      newPos = opEnd;
    }

    pos = newPos;
  });

  return secondsToTime(pos);
}

function timeToSeconds(time) {
  const [minutes, seconds] = time.split(":");
  return Number(minutes) * 60 + Number(seconds);
}

function secondsToTime(seconds) {
  return `${String(Math.floor(seconds / 60)).padStart(2, "0")}:${String(
    seconds % 60
  ).padStart(2, "0")}`;
}
